sdk/src/types/block/core/basic.rs

```diff
@@ -95,6 +118,75 @@ impl BasicBlockBuilder {
     }
 }
 
+impl BlockBuilder<BasicBlockBuilder> {
```

Comment:
Honestly I feel like this is a remnant from the previous version where the wrapper itself was generic. I don't think we need generics for the builder and I even think it makes it more complex and duplicates code. We already have a `BasicBlockBuilder`, we don't need more. We probably just need to add a very simple and straightforward BlockWrapperBuilder.

Reply:
In that case we would also need a `BlockBuilder` enum and everything that comes with that

Reply:
Hmm not sure about that, if the wrapper builder takes an already built block I don't see why we'd need an enum. But I haven't tried it so maybe I'm missing something. I can have a look tomorrow.

Reply:
I went ahead and removed it